### DISCUSSION TITLE
AnimateDiff

### DISCUSSION THREAD

**Discussion by vladmandic** [vladmandic/sdnext#2396] [General]
Just sharing my notes...

- minimal supported version of **sd.next** is 10/17/2023
- must use *backend:original*
- install [animatediff](https://github.com/continue-revolution/sd-webui-animatediff) extension as usual  
- download animatediff models and loras as usual  
  *note: only _v2 models allow usage of animatediff loras, for example `mm_sd_v15_v2`*

and thats it.  

my recommended workflow is slightly different as animatediff does not do a great job with interpolation.  
not to mention it creates a dependency on `deforum` if you just want to create mp4.  

- settings:  
  - settings -> image options: enable "always save all generated images"  
  - settings -> image naming: enable "add number to filename when saving"  
  - settings -> animatediff: enable "calculate the optimal GIF palette"  
  - animatediff extension panel: select PNG, unselect GIF and MP4, frame interpolation off  
- use animatediff to create sequence of images  
- use [rife](https://github.com/vladmandic/rife) to create actual movie  
  much better results with frame interpolation video stabilization  
  example: `python rife.py --input outputs/text/ --mp4 --seq 4955 --rm`  
  full info: `python rife.py --help`

**Comment by finefin** [comment]
I never got this extension working :(
Today I tried again: did a completely new install of SDNext and installed the animatediff extension - still get a lot of errors:
```
Module load: extensions\sd-webui-animatediff\scripts\animatediff_ui.py: ImportError
┌────────────────────────────────────────────────── Traceback (most recent call last) ───────────────────────────────────────────────────┐
│ E:\AI\automatic-fork\SDNext\automatic\modules\script_loading.py:21 in load_module                                                      │
│                                                                                                                                        │
│   20 │   │   │   with contextlib.redirect_stdout(io.StringIO()) as stdout:                                                             │
│ > 21 │   │   │   │   module_spec.loader.exec_module(module)                                                                            │
│   22 │   │   │   setup_logging() # reset since scripts can hijaack logging                                                             │
│ in exec_module:883                                                                                                                     │
│ in _call_with_frames_removed:241                                                                                                       │
│                                                                                                                                        │
│ E:\AI\automatic-fork\SDNext\automatic\extensions\sd-webui-animatediff\scripts\animatediff_ui.py:10 in <module>                         │
│                                                                                                                                        │
│     9 from scripts.animatediff_mm import mm_animatediff as motion_module                                                               │
│ >  10 from scripts.animatediff_i2ibatch import animatediff_i2ibatch                                                                    │
│    11                                                                                                                                  │
│                                                                                                                                        │
│ E:\AI\automatic-fork\SDNext\automatic\extensions\sd-webui-animatediff\scripts\animatediff_i2ibatch.py:11 in <module>                   │
│                                                                                                                                        │
│    10 from modules import processing, shared, scripts, img2img, devices, masking, sd_samplers,                                         │
│ >  11 from modules.processing import (StableDiffusionProcessingImg2Img,                                                                │
│    12 │   │   │   │   │   │   │   │   process_images,                                                                                  │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
ImportError: cannot import name 'images_tensor_to_samples' from 'modules.processing'
(E:\AI\automatic-fork\SDNext\automatic\modules\processing.py)
```

EDIT: no idea what I'm doing wrong or what I'm missing. I guess I'll stop trying and do something else :D

**Reply by bluekght** [reply]
Same boat here, different error.

For what it's worth, I got AnimateDiff working with ComfyUI, you may consider giving that a try if you really need it? Although who knows, if the AnimateDiff author keeps changing how it works, they might break that too.

**Comment by vladmandic** [comment]
its not "still", its "again" :(
animatediff keeps changing how it hooks up into core app almost daily and its causing frequent breakages.
i think i fixed this one, but there's a limit how much i'm wiling to spend to support animatediff.

**Comment by vladmandic** [comment]
Latest fix was in sdnext dev branch only as I can't mess with master each time something changes.
Dev merge to master is scheduled for end of this week.

**Reply by finefin** [reply]
thank you <3 let's see how long it'll work.
